packages/firestore/src/local/indexeddb_schema.ts

```diff
@@ -172,6 +179,49 @@ export class SchemaConverter implements SimpleDbSchemaConverter {
       });
     });
   }
+
+  /**
+   * Ensures that every document in the remote document cache has a corresponding sentinel row
+   * with a sequence number. Missing rows are given the most recently used sequence number.
+   */
+  private ensureSequenceNumbers(
+    txn: SimpleDbTransaction
+  ): PersistencePromise<void> {
+    const documentTargetStore = txn.store<
+      DbTargetDocumentKey,
+      DbTargetDocument
+    >(DbTargetDocument.store);
+    const documentsStore = txn.store<DbRemoteDocumentKey, DbRemoteDocument>(
+      DbRemoteDocument.store
+    );
+
+    return getHighestListenSequenceNumber(txn).next(currentSequenceNumber => {
+      const writeSentinelKey = (
```

Comment:
I think this would be easier to read if a) this was declared one level higher (as a direct child of `ensureSequenceNumbers`) and b) if you used a normal JS function (`function writeSentinelKey() {}`).

Optional.

Reply:
Leaving as-is.

It can't go in a higher scope as it closes over the result of `getHighestListenSequenceNumber()`. And typescript disallows nested `function` declarations, so it needs to be a lambda. I'm guessing this is because `function` and lambdas have different bindings for `this`. Nested `function`s make it easy to make a mistake when calling methods off of `this`.